### ISSUE TITLE
[BUG] Stack smashing detected on RTX 4090 after point cloud initialization

### ISSUE THREAD

**Issue by zjkccb** [xiaobiaodu/Mobile-GS#10]
## Environment
- **OS**: Ubuntu 22.04 (clean install)
- **GPU**: NVIDIA RTX 4090 (24GB)
- **Driver**: 560.28.03
- **CUDA Toolkit**: 11.8 (installed via `conda install -c nvidia/label/cuda-11.8.0 cuda-toolkit`)
- **PyTorch**: 2.5.1+cu118
- **Python**: 3.11
- **GCC**: tried both 11.4.0 and 9.5.0 (specified via `CC`/`CXX` during compilation)
- **Other dependencies**: installed as per `requirements.txt`, plus manually added `icecream`, `matplotlib`, etc.

## Steps to Reproduce
1. Clone the repository.
2. Create and activate conda environment with Python 3.11.
3. Install PyTorch 2.5.1 for CUDA 11.8.
4. Install submodules with `--no-build-isolation` and `TORCH_CUDA_ARCH_LIST="8.9"`.
5. Download Mip-NeRF 360 dataset (tested with `bicycle` and `garden` scenes).
6. Run:
   ```bash
   python pretrain.py -s data/bicycle -m output/test --eval --imp_metric outdoor --sh_degree 1 --iterations 100

Observed Behavior
The script loads the dataset and cameras successfully, converts point clouds, and prints:
Number of points at initialisation :  54275

Immediately after this line, the program crashes with:
*** stack smashing detected ***: terminated
Aborted (core dumped)

Expected Behavior
Training should start and show iteration progress.

Attempted Fixes (None Resolved the Issue)
Recompiled all submodules (diff-gaussian-rasterization_ms, diff-gaussian-rasterization_ms_nosorting, simple-knn) with TORCH_CUDA_ARCH_LIST="8.9" for RTX 4090.

Set ulimit -s unlimited to increase stack size.

Added -fno-stack-protector to CFLAGS and CXXFLAGS.

Tried both GCC-11 and GCC-9 as host compiler.

Tested on a smaller scene (bicycle vs garden) and with reduced sh_degree=1.

Added torch.cuda.set_limit(torch.cuda.limit_stack_size, 16384) in the Python script.

The crash occurs consistently at the same point.

Additional Context
The issue strongly suggests a buffer overflow in one of the CUDA kernels, likely due to fixed-size stack arrays that are insufficient for RTX 4090's warp/thread configuration. Would you be able to provide guidance on patching the relevant .cu files, or is there an existing fix?

Thank you for your time and for this great project!

**Comment by xiaobiaodu** [comment]
Hi. Please try our improved version of Mobile-GS, namely Flux-GS https://github.com/xiaobiaodu/Flux-GS
It would be better than Mobile-GS
